### PR TITLE
Assorted nits from C# code.

### LINKED ISSUE #535: Add documentation to ByteString.CopyFrom
When the parameter to ByteString.CopyFrom is a `params byte[]`, we should give an example of why that's useful (primarily for testing).

### PR DESCRIPTION
**Pull request by jtattermusch**
Fixes #535.

### CODE REVIEW

File: csharp/src/ProtocolBuffers/InvalidProtocolBufferException.cs [hunk 1]
```diff
@@ -79,12 +79,6 @@ internal static InvalidProtocolBufferException InvalidEndTag()
                 "Protocol message end-group tag did not match expected tag.");
         }
 
-        internal static InvalidProtocolBufferException InvalidWireType()
```
Comment (jtattermusch) [inline csharp/src/ProtocolBuffers/InvalidProtocolBufferException.cs]: Looks like this was unused. Remove it?

File: csharp/src/ProtocolBuffers.Test/Collections/RepeatedFieldTest.cs [hunk 2]
```diff
@@ -239,21 +239,19 @@ public void Clone_ReturnsMutable()
         public void Enumerator()
         {
             var list = new RepeatedField<string> { "first", "second" };
-            using (var enumerator = list.GetEnumerator())
```
Comment (jskeet) [inline csharp/src/ProtocolBuffers.Test/Collections/RepeatedFieldTest.cs]: No, this *should* be okay - and we *should* dispose of iterators. That's what foreach will do, for example. We should look more carefully into what's wrong here...

**Comment by jskeet** [comment]
I'm happy with all but removing the using statement. Will discuss that separately.

**Comment by jtattermusch** [comment]
Please look at #603 to find out some intesting things about the using statement and enumerator.

File: csharp/src/ProtocolBuffers.Test/Collections/RepeatedFieldTest.cs [hunk 3]
```diff
@@ -239,21 +239,19 @@ public void Clone_ReturnsMutable()
...
-                Assert.AreEqual("second", enumerator.Current);
-                Assert.IsFalse(enumerator.MoveNext());
-                Assert.Throws<InvalidOperationException>(() => enumerator.Current.GetHashCode());
-                Assert.IsFalse(enumerator.MoveNext());
-                enumerator.Reset();
-                Assert.Throws<InvalidOperationException>(() => enumerator.Current.GetHashCode());
-                Assert.IsTrue(enumerator.MoveNext());
-                Assert.AreEqual("first", enumerator.Current);
-            }
+            var enumerator = list.GetEnumerator();
```
Comment (jskeet) [inline csharp/src/ProtocolBuffers.Test/Collections/RepeatedFieldTest.cs]: Ah, having read http://stackoverflow.com/questions/4642665 I see the issue.
I'll have a think about it. Another option could be to use a try/finally block manually, which might fix it.

Will certainly look at whether the mutable struct provides any benefit any more within the runtime - it still would for clients, in normal situations. (The capture here isn't particularly "normal"...)

**Comment by jtattermusch** [comment]
Rebased this one to not contain the fix from #603. Please merge this once happy.

**Pull request merged**